dexbot/controllers/main_controller.py

```diff
@@ -41,6 +41,24 @@ def __init__(self, bitshares_instance, config):
         # Initialize folders
         initialize_data_folders()
 
+    def set_bitshares_instance(self, bitshares_instance):
+        """ Set bitshares instance
+
+            :param bitshares_instance: A bitshares instance
+        """
+        self.bitshares_instance = bitshares_instance
+        set_shared_bitshares_instance(bitshares_instance)
+
+    def new_bitshares_instance(self, node, retries=-1, expiration=60):
+        """ Create bitshares instance
+
+            :param retries: Number of retries to connect, -1 default to infinity
```

Comment:
if the end user is not connected to the internet this will spin forever. perhaps put a time limit on it, like 60 seconds? or 90 seconds?  Notify end user in GUI they need to reconnect?